share/revng/include/primitive-types.h

```diff
@@ -166,7 +166,8 @@ extern uint128_t undef_uint128_t();
 // Float
 //
 
-#if __ARM_FP16_ARGS == 1 || defined(__FLT16_MIN__)
+#if (__ARM_FP16_ARGS == 1 || defined(__FLT16_MIN__)) \
+  && !defined(DISABLE_FLOAT16)
```

Comment:
I'd prefer to hoist the `DISABLE_XXX` macros at the top, as a form of documentation, as in my original proposal.